core_backend/tests/api/test_TTS.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from core_backend.app.llm_call.llm_prompts import IdentifiedLanguage
+from core_backend.app.voice_api.text_to_speech import generate_speech
+
+
+@pytest.fixture
+def mock_gtts() -> MagicMock:
+    mock_gTTS = MagicMock()
+    mock_gTTS_instance = mock_gTTS.return_value
+    mock_gTTS_instance.save = MagicMock()
+    return mock_gTTS
+
+
+class TestGenerateSpeech:
+    @pytest.mark.parametrize(
+        "language, text",
+        [
+            (IdentifiedLanguage.ENGLISH, "Hello, this is a test."),
+            (IdentifiedLanguage.SWAHILI, "Habari, hii ni jaribio."),
+            (IdentifiedLanguage.HINDI, "नमस्ते, यह एक परीक्षण है।"),
+        ],
+    )
+    @pytest.mark.asyncio
+    async def test_generate_speech_success(
+        self, language: IdentifiedLanguage, text: str, mock_gtts: MagicMock
+    ) -> None:
+        save_path = "test_audio.mp3"
+
+        with patch("core_backend.app.voice_api.text_to_speech.gTTS", mock_gtts):
+            tts_file_path = await generate_speech(text, language, save_path)
+            assert tts_file_path == save_path
+
+    @pytest.mark.parametrize(
+        "text, language",
```

Comment:
You should add another use case when you type gibberish and the language is supported. Also I think it's worth checking when the text is in a language (like hindi) and the specified language is different (for ex Swahili) to see the behaviour.

Reply:
it would not raise any error right now for these specific test cases cause: 
1) as the TTS is being used fairly late in the workflow (after we get a response from the LLM) the guardrails would already catch errors and not let the TTS file get generated at all.
2) as the external TTS we using right now is GTTS, it is limited in it capabilities and cannot identify if the langauge would be gibberish or there is a language mismatch.

but i think that should not be an issue right as when used with the rest of the workflow and with the necessary guardrails implemented (as we would mostly get the correct language identified and it wouldnt give an answer to gibberish) these wouldnt pose a risk, but we can still try implementing extra error handling in the inhouse TTS

Reply:
It makes sense not to test TTS doing that on question_answer, but here we are testing the TTS function alone, so We should make sure the function is robust enough to catch these issues.

Reply:
so what do u suggest, right now the external API is not robust enough to catch these errors but when we replace it with an actual neural model we could catch these errors at that time so for now should i merge it?

Reply:
If the api don't catch it then the test wont pass. You can merge for now